src/main/java/world/bentobox/warps/Warp.java

```diff
@@ -126,7 +127,24 @@ public PluginConfig getSettings() {
      * @return island level or null if there is no level plugin
      */
     public Long getLevel(World world, UUID uniqueId) {
-        return plugin.getAddonsManager().getAddonByName(LEVEL_ADDON_NAME).map(l -> ((Level) l).getIslandLevel(world, uniqueId)).orElse(null);
+
+        Optional<Addon> levelHook = this.getAddonByName(LEVEL_ADDON_NAME);
+
+        if (levelHook.isPresent())
+        {
+            Object value = new AddonRequestBuilder().addon(LEVEL_ADDON_NAME).
+                label("island-level").
+                addMetaData("player", uniqueId).
+                addMetaData("world-name", world.getName()).
+                request();
+
+            if (value != null)
+            {
+                return (Long) value;
+            }
+        }
+
+        return null;
```

Comment:
Using the addon request API is not needed here. Addons can access other addons's methods and classes.

Reply:
:D
I used addon request API as it is much easier to get data from any addon, as it is not necessary to import it as dependency.